pyproject.toml

```diff
@@ -2,3 +2,30 @@
 line-length = 120
 skip-string-normalization = true
 target-version = ['py37']
+
+[tool.pylint.main]
+init-hook = 'import sys; sys.path.append("./src")'
+ignore-patterns = ["test/", "src/bin"]
+load-plugins = ["pylint.extensions.bad_builtin", "pylint.extensions.mccabe"]
+extension-pkg-allow-list = ["yara"]
+jobs = 0
+
+[tool.pylint."messages control"]
+confidence = []
+disable = ["missing-docstring" , "locally-disabled", "logging-format-interpolation", "logging-fstring-interpolation"]
+
+
+[tool.pylint.reports]
+output-format = "colorized"
+msg-template="{path}:{line}: [{symbol}:{obj}] {msg}"
+
+[tool.pylint.format]
+max-line-length = 120
+
+[tool.pylint.design]
```

Comment:
From reading I'd say you two can decide if two-letter-variables should be valid or not (they occur quite regularly in our code I think).
Otherwise I'd propose to go with the more _default_ configuration and only reintroduce changes if need be.
Defaults with these things are most often a sensible choice.

Reply:
>There are still some differences which I noticed when comparing the results for both configs on some files:

Have a look at the commit message. I tried to document everything that I changed.

> `max-args=7`

Whoots yeah slight oversight.
Luckily we don't have any function with more than 7 arguments (`rg -g "*.py" 'def .+?\(.+?,.+?,.+?,.+?,.+?,.+?,.+?,.+?\)'`).
We have two functions with 7 arguments and several with 6.
So lets leave this as it is.

> but e.g. variable names with length 2 like fp or fo are no longer legal with the default patterns)
I agree that it does not make much sense to maintain our own regexes just for having to letter variables.
There is `good-names` which allows us to define names that pass the check even if they don't match the default regex.
See the latest commit.